src/main/java/hudson/tasks/test/AbstractTestResultAction.java

```diff
@@ -509,4 +1593,46 @@ public Object readResolve() {
         }
     }
 
+    /**
+     * A generic class to encapsulate a pair of objects where each object can be any type.
+     * @param <A> Type of the first object.
+     * @param <B> Type of the second object.
+     */
+    private static final class Pair<A,B>{
+
+        /**
+         * First object.
+         */
+        A first;
+
+        /**
+         * Second object.
+         */
+        B second;
+
+        /**
+         * A constructor to create a {@link Pair} object containing both the objects
+         * {@link Pair#first} and {@link Pair#second} as pair.
+         * @param first The first object conatined in the pair.
+         * @param second The second object contained in the pair.
+         */
+        Pair(A first, B second){
+            this.first = first;
+            this.second = second;
+        }
+    }
+
+    private static final String ISFAILUREONLY = "false";
+    private static final String ALLPROJECTS = "AllProjects";
+    private static final String TREND1 = "BuildAnalysis";
```

Comment:
Naming can be improved. something like LENGTH_TESTS_TREND instead of TREND1 would make code more readable.

Reply:
I'll do that. Thanks.